keras_nlp/tokenizers/unicode_character_tokenizer.py

```diff
@@ -0,0 +1,311 @@
+                '"NFD", "NFKD". Received: normalization_form='
+                f"{normalization_form}"
+            )
+
+        # Check errors.
+        if errors not in ["strict", "replace", "ignore"]:
+            raise ValueError(
+                '`errors` must be one of "strict", "replace", "ignore" '
+                f"Received: errors={errors}"
+            )
+
+        super().__init__(**kwargs)
+
+        self.sequence_length = sequence_length
+        self.lowercase = lowercase
+        self.normalization_form = normalization_form
+        self.errors = errors
+        self.replacement_char = replacement_char
+        self.input_encoding = input_encoding
+        self.output_encoding = output_encoding
+
+    def get_config(self) -> Dict[str, Any]:
+        config = super().get_config()
+        config.update(
+            {
+                "sequence_length": self.sequence_length,
+                "lowercase": self.lowercase,
+                "normalization_form": self.normalization_form,
+                "errors": self.errors,
+                "replacement_char": self.replacement_char,
+                "input_encoding": self.input_encoding,
+                "output_encoding": self.output_encoding,
+            }
+        )
+        return config
+
+    def tokenize(self, inputs):
+
+        if not isinstance(inputs, (tf.Tensor, tf.RaggedTensor)):
+            inputs = tf.convert_to_tensor(inputs)
+
+        scalar_input = inputs.shape.rank == 0
+        if scalar_input:
+            inputs = tf.expand_dims(inputs, 0)
+
+        # Optionally Lowercase the Text
+        if self.lowercase:
+            inputs = tf_text.case_fold_utf8(inputs)
+
+        # Optionally Normalize the Text to a given form
+        if self.normalization_form:
+            if self.input_encoding != "utf8":
+                raise ValueError(
+                    """Normalization Forms are Only Supported for Input Encoding
+                     utf-8"""
+                )
+            else:
+                inputs = tf_text.normalize_utf8(inputs, self.normalization_form)
+
+        # Apply Unicode Decoder
+        tokens = tf.strings.unicode_decode(
+            inputs,
+            errors=self.errors,
+            replacement_char=self.replacement_char,
+            input_encoding=self.input_encoding,
+        )
+
+        if self.sequence_length:
+            output_shape = tokens.shape.as_list()
+            output_shape[-1] = self.sequence_length
+            tokens = tokens.to_tensor(shape=output_shape)
+
+        if scalar_input:
+            tokens = tf.squeeze(tokens, 0)
+        return tokens
+
+    def detokenize(self, inputs):
+        encoded_string = tf.strings.unicode_encode(
+            inputs,
+            errors=self.errors,
+            replacement_char=self.replacement_char,
+            output_encoding=self.output_encoding,
+        )
+        encoded_string_with_removed_null = tf.strings.regex_replace(
```

Comment:
Do this before the unicode_encode call with `tf.ragged.boolean_mask(inputs, inputs != 0)`

Reply:
Thanks for the review!
I'm not entirely sure why do we even require this mask? Also since `regex_replace` works on strings I don't believe it can be done before the unicode_encode call. Am I missing something?

Reply:
`tf.ragged.boolean_mask(inputs, inputs != 0)` would replace this like entirely, by removing all the zeros from the input before unicode encoding, rather than after. I think it is more readable, as regexes are often hard to parse. See ByteTokenizer as an example.